packages/core/src/storage.ts

```diff
@@ -217,5 +260,13 @@ export interface Storage {
     query: DocumentQuery<T>,
     options?: FindOptions<T>
   ) => Promise<FindResult<T>>
+
   tx: (tx: Tx) => Promise<TxResult>
 }
+
+/**
+ * @public
+ */
+export interface FulltextStorage {
+  searchFulltext: (query: SearchQuery, options: SearchOptions) => Promise<SearchResult>
```

Comment:
Should we design it with some streaming capabilities? Like we start search, receive some results and immediately display them, and send more requests in context like to have more items of particular type, or if some more data is arriving?

Reply:
It would be interesting. I assume you are talking about Scroll API. I never saw using of it in user-facing search scenarios, only for large amounts of data such as logs or something. So I don't see any usage of it for mentions and spotlight, but if you could give some clue, some usage example for our system, I would be happy to build such API.